src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -118,6 +118,13 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
   }
 
+  toPreviousPage() {
+    let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
+    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
+    this.updateFormData(caseEventData as CaseEventData);
+    this.previous();
+  }
+
```

Comment:
may be a good unit test to cover this logic?

Reply:
I can add another test that evaluate the call to comp.updateFormData(jsonData);  and this.previous ()